src/main/scala/mesosphere/marathon/health/HealthCheck.scala

```diff
@@ -122,4 +122,36 @@ object HealthCheck {
   val DefaultMaxConsecutiveFailures = 3
   val DefaultIgnoreHttp1xx = false
   val DefaultPort = None
+
+  implicit val healthCheck = validator[HealthCheck] { hc =>
+    (hc.portIndex.nonEmpty is true) or (hc.port.nonEmpty is true)
+    hc is validProtocol
+  }
+
+  /**
```

Comment:
Regarding your second remark: do you mean the *comment* should be moved to appDefintionValidator?